packages/core/src/types/languages.ts

```diff
@@ -89,14 +84,15 @@ export interface LanguageFile extends Disposable {
 export interface LanguageFileDefinition extends Partial<Disposable> {
 	cache?: LanguageFileCacheImpacts;
 	getDiagnostics?(): LanguageDiagnostics;
+	normalizeRange(range: CharacterReportRange): NormalizedReportRangeObject;
 	runRule<
 		OptionsSchema extends AnyOptionalSchema | undefined =
 			| AnyOptionalSchema
 			| undefined,
 	>(
-		rule: AnyRuleDefinition<OptionsSchema>,
+		ruleRuntime: AnyRuleRuntime<InferredObject<OptionsSchema>>,
```

Comment:
I do still think we should thread the generics through here (it caught a few issues IIRC), but we can always save that for a different PR as well.